src/topology/algebra/infinite_sum.lean

```diff
@@ -887,6 +899,30 @@ lemma tsum_comm [regular_space α] {f : β → γ → α} (h : summable (functio
   ∑' c b, f b c = ∑' b c, f b c :=
 tsum_comm' h h.prod_factor h.prod_symm.prod_factor
 
+/-- Let `f : ℕ → ℝ` be a sequence with summable series and let `i ∈ ℕ` be an index.
+Lemma `tsum_ite_eq_extract` writes `Σ f n` as the sum of `f i` plus the series of the
+remaining terms. -/
+lemma tsum_ite_eq_extract {f : ℕ → ℝ} (hf : summable f) (i : ℕ) :
+  ∑' n, f n = f i + ∑' n, ite (n ≠ i) (f n) 0 :=
+by rw [← tsum_ite_eq_add (hf.summable_of_eq_zero_or_self (λ j, _)) i, tsum_congr (λ j, _)];
+  by_cases ji : j = i; simp [ji]
+
+/-- Let `f, g : ℕ → ℝ` be two sequences with summable series.  If `f` is dominated by `g` and
+at least one term of `f` is strictly smaller than the corresponding term in `g`, then the series
+of `f` is strictly smaller than the series of `g`. -/
+lemma tsum_lt {f g : ℕ → ℝ} (h : ∀ (b : ℕ), f b ≤ g b)
+  (hf: summable f) (hg: summable g) {i : ℕ} (hi : f i < g i) :
+  ∑' n, f n < ∑' n, g n :=
+begin
+  rw [tsum_ite_eq_extract hf i, tsum_ite_eq_extract hg i],
+  refine add_lt_add_of_lt_of_le hi (tsum_le_tsum (λ j, _) _ _),
+  by_cases ji : j = i; simp [ji]; exact h j,
+  { refine hf.summable_of_eq_zero_or_self (λ j, _),
+    by_cases ji : j = i; simp [ji] },
+  { refine hg.summable_of_eq_zero_or_self (λ j, _),
+    by_cases ji : j = i; simp [ji] }
```

Comment:
I think you should extract a lemma that says `summable (∑' b', ite (b' = b) 0 (f b'))`

Reply:
Or better, the statement that:
```
lemma has_sum.update (fs : α) (hf : has_sum f fs) (b : β) [decidable_eq β] (a : α) :
  has_sum (update f b a) (a - f b + fs) :=
begin
  convert ((has_sum_ite_eq b _).add hf),
  ext b',
  by_cases h : b' = b,
  { rw h, simp, },
  { simp [h] },
end
```

Reply:
With that, this proof becomes
```
/-- Let `f, g : ℕ → ℝ` be two sequences with summable series.  If `f` is dominated by `g` and
at least one term of `f` is strictly smaller than the corresponding term in `g`, then the series
of `f` is strictly smaller than the series of `g`. -/
lemma tsum_lt {f g : ℕ → ℝ} (h : ∀ (b : ℕ), f b ≤ g b)
  (hf: summable f) (hg: summable g) {i : ℕ} (hi : f i < g i) :
  ∑' n, f n < ∑' n, g n :=
begin
  have := has_sum_le (λ j, _) (hf.has_sum.update i 0) (hg.has_sum.update i 0),
  simpa using add_lt_add_of_lt_of_le hi this,
  by_cases hij : j = i,
  { rw hij, simp, },
  { simp [update_noteq hij, h j] },
end
```

Reply:
I am happy to follow your suggestion, however, to me the proof does not look easier.  Why do you prefer this version?  I am only asking to understand which arguments are preferrable to which other ones.

Reply:
Well, `has_sum.update` seems like a useful lemma to have, since `update` is used quite often in mathlib. I don't feel too strongly about whether you use it to prove this lemma, but it saves you the repeated branches of proofs.

Reply:
Ah, I think the real takeaway here is that it's always better to write the lemma in terms of `has_sum` or `summable` instead of `tsum`, as any time you use lemmas about the latter you end up having to prove the former anyway.

Reply:
Ok, personally, I am ok with the proofs as they are now.  I had tried playing around with `has_sum` but gave up, since I had a little of a hard time.

If you really think that it would be worth adding the API for the `has_sum` part, I may try it again, but I may need some further help with that!